lifecycle/launch/lifecycle_demo_launch.py

```diff
@@ -12,24 +12,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from launch.legacy.exit_handler import primary_exit_handler
-from ros2run.api import get_executable_path
+from launch import LaunchDescription
+import launch.actions
+from launch_ros.substitutions import ExecutableInPackage
 
 
-def launch(launch_descriptor, argv):
-    ld = launch_descriptor
-
-    package = 'lifecycle'
-    ld.add_process(
-        cmd=[get_executable_path(package_name=package, executable_name='lifecycle_talker')],
-    )
-
-    ld.add_process(
-        cmd=[get_executable_path(package_name=package, executable_name='lifecycle_listener')],
-    )
-
-    ld.add_process(
-        cmd=[get_executable_path(
-            package_name=package, executable_name='lifecycle_service_client')],
-        exit_handler=primary_exit_handler,
-    )
+def generate_launch_description():
+    return LaunchDescription([
+        launch.actions.ExecuteProcess(
+            cmd=[ExecutableInPackage(package='lifecycle', executable='lifecycle_talker')],
```

Comment:
I think you can/should use the `launch_ros.actions.LifecycleNode` action for the `lifecycle_talker` and `lifecycle_listener` instead and the `launch_ros.actions.Node` for the `lifecycle_service_client`.

Reply:
Can `launch_ros.actions.LifecycleNode` be used on regular rclcpp Nodes ?

If it doesn't make sense to do so, we should likely not use it for the listener as it is a "regular node" AFAICT: https://github.com/ros2/demos/blob/c9f8035d2282d4f553b28ac786efcb55742953cd/lifecycle/src/lifecycle_listener.cpp#L34

Reply:
If it is not an actual lifecycle node then it will work, but it will have unnecessary overhead and if someone attempts to make launch change its state that will fail).

So I'd say you're right we should use `Node` on the confusingly name :p `lifecycle_listener`.